Week5/app/src/main/java/com/em4n0101/mymovies/repositories/MoviesRepository.kt

```diff
@@ -0,0 +1,49 @@
+package com.em4n0101.mymovies.repositories
+
+import android.app.Application
+import android.os.AsyncTask
+import androidx.lifecycle.LiveData
+import com.em4n0101.mymovies.data.Movie
+import com.em4n0101.mymovies.database.MoviesDatabase
+import com.em4n0101.mymovies.database.MoviesDatabaseDao
+
+class MoviesRepository(application: Application) {
+    private val moviesDao: MoviesDatabaseDao = MoviesDatabase.getInstance(application).moviesDatabaseDao
+
+    fun insert(movie: Movie) {
+        InsertAsyncTask(moviesDao).execute(movie)
+    }
+
+    fun getMovies(): LiveData<List<Movie>> {
+        return moviesDao.getAllMovies()
+    }
+
+    fun getMovieBy(title: String): LiveData<Movie> {
+        return moviesDao.get(title)
+    }
+}
+
+private class InsertAsyncTask(private val moviesDao: MoviesDatabaseDao):
```

Comment:
I know AsyncTask was used in the course, but it's deprecated now and try not to use it in the future.

Reply:
It's just bad api, and often causes bugs to appear